helpers/withApollo.ts

```diff
@@ -16,6 +16,8 @@ export default withApollo(
       // Cache is used to rehydrating the store for Server Side Rendering.
       // Apollo Explanation https://www.apollographql.com/docs/react/performance/server-side-rendering/
       // Example from : https://medium.com/swlh/create-a-killer-frontend-for-2020-setup-next-js-graphql-styled-components-typescript-and-ssr-fe66cffd7d94
+      //next-with-appollo doesn't support apollo3 yet, so some types are not up to date
+      //@ts-ignore
```

Comment:
next-with-apollo wasn't updated for apollo 3 yet. It works fine but some types are wrong, hence ignore.

Reply:
Wouldn't it better to wait until it's official supported before we use in prod? Implementing new library updates early means that we're doing the testing for other people. 

It may not be a good idea for new students to be exposed to this side of development.

Reply:
It is possible to remove `next-with-apollo` completely and use next.js fetching methods with graphql as mentioned in the linked issue but it will require a lot of work. On the other hand, `next-with-apollo` only uses apollo for types and tests [according](https://github.com/lfades/next-with-apollo/pull/132#issuecomment-702773588) to its creator.

Reply:
That is fine. I am not asking to remove `next-with-apollo`. 

I think it's better to delay the implementation with apollo3 until `next-with-apollo` is updated so we don't have any workaround.  Are we getting any benefit out of migrating to apollo3?

Reply:
`next-with-apollo` won't be getting a update any time soon I think, since there's a PR opened 9 months ago to support Apollo 3. Even the maintainer, who works at Vercel, recommends using the next.js fetch methods now instead of the module HOC.

Besides that I believe we would just be delaying the inevitable as things move fast nowadays. Maybe it's better to be up to date now or we might need to spend more time migrating stuff later to support new features.

Would like to hear everyone's opinion on this.

Reply:
It would have been more clear if the benefits were explained from the start:

- Apollo Client 3 has already been out for six months and `next-with-apollo` has no priority in being replaced soon. It would be worth to implement client 3 due to the improvements below from Apollo 3
- The app can be refactored to import from one package. It makes it easier for new users to understand.
- The improvements to the cache makes implementation a lot more clean and way less cumbersome. 

For next time, when requesting to upgrade to a new version, it would be helpful to list the benefits so other people can understand the need to go to a new version.

Reply:
Yeah that's a fair point, will keep that in mind 👍

Reply:
@songz